src/compiler/getAppSource.ts

```diff
@@ -1,17 +1,25 @@
 import { promises as fs } from 'fs';
-import { resolve, relative } from 'path';
+import { resolve, relative, join } from 'path';
 import { IAppInfo } from '@rocket.chat/apps-engine/definition/metadata';
+import { CompilerOptions } from 'typescript';
 
 import { IAppSource, ICompilerFile, IMapCompilerFile } from '../definition';
+import { AppCompilerOptions } from '../AppsCompiler';
+import logger from '../misc/logger';
 
-async function walkDirectory(directory: string): Promise<ICompilerFile[]> {
+export type TSConfig = {
+    compilerOptions?: CompilerOptions;
+    exclude?: string[];
+}
+
+async function walkDirectory(directory: string, projectExcludes: string[] = []): Promise<ICompilerFile[]> {
     const dirents = await fs.readdir(directory, { withFileTypes: true });
+    const dirsToIgnore = projectExcludes.concat(['node_modules', '.git']);
```

Comment:
Can this be included in app.json file? Also, can we add files here too?

Reply:
That's the idea for the `projectExcludes` parameter